ngraph/frontend/onnx_common/CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+# Copyright (C) 2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+#
+
+set(TARGET_NAME "onnx_common")
+
+file(GLOB_RECURSE LIBRARY_SRC ${CMAKE_CURRENT_SOURCE_DIR}/src/*.cpp)
+file(GLOB_RECURSE PUBLIC_HEADERS ${CMAKE_CURRENT_SOURCE_DIR}/include/*.hpp)
```

Comment:
using `GLOB_RECURSE` for source files is not recommended in cmake documentation:
https://cmake.org/cmake/help/latest/command/file.html#glob-recurse

Note
We do not recommend using GLOB to collect a list of source files from your source tree. If no CMakeLists.txt file changes when a source is added or removed then the generated build system cannot know when to ask CMake to regenerate. The CONFIGURE_DEPENDS flag may not work reliably on all generators, or if a new generator is added in the future that cannot support it, projects using it will be stuck. Even if CONFIGURE_DEPENDS works reliably, there is still a cost to perform the check on every rebuild.

Reply:
As before, I see your argument and I know that it is rather described as anti-pattern.
@ilyachur What do you think?

Reply:
I understand your concerns about `GLOB_RECURSE` but here I have the same comment as above. At the current moment we are using `GLOB_RECURSE` for all OpenVINO components.
I would like to have the one approach to all components.

Reply:
So maybe we should start the change?
It will be better if other components will switch to recommended solution instead of using GLOBS.
We should rather go in the right direction instead of keeping the old one only because it is commonly used.

Reply:
I don't think that it is good idea because in some targets we have hundreds of files it will be a problem to enumerate all these files.
IMHO, the cmake regeneration is not a big problem.